x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -1007,24 +1019,29 @@ export const RulesList = ({
       )}
     </>
   );
-  // if initial load, show spinner
-  const getRulesList = () => {
-    if (noData && !rulesState.isLoading && !ruleTypesState.isLoading) {
-      return authorizedToCreateAnyRules ? (
-        <EmptyPrompt
-          showCreateRuleButton={showCreateRuleButton}
-          onCTAClicked={() => setRuleFlyoutVisibility(true)}
-        />
-      ) : (
-        noPermissionPrompt
-      );
-    }
 
+  const showPrompt = noData && !rulesState.isLoading && !ruleTypesState.isLoading;
+
+  useEffect(() => {
     if (initialLoad) {
-      return <CenterJustifiedSpinner />;
+      return;
     }
+    if (!showPrompt && authorizedToCreateAnyRules) {
+      setHeaderActions?.([<CreateRuleButton openFlyout={openFlyout} />, <RulesListDocLink />]);
+    } else {
+      setHeaderActions?.();
+    }
+  }, [initialLoad, showPrompt, authorizedToCreateAnyRules]);
+
+  useEffect(() => {
+    return () => setHeaderActions?.();
```

Comment:
Mostly for defensive reasons, in case we decide to add another tab, we wouldn't want the old headers to appear in the new tab. Especially because the rules list has a "create rule" button, which is only relevant for the rules list